packages/react-server/index.ts

```diff
@@ -1,47 +1,63 @@
 import {
   Fragment,
   createElement,
-  forwardRef,
   memo,
   useCallback,
   useEffect,
   useRef,
   useState,
 } from 'react';
-import type { ComponentType, ForwardedRef } from 'react';
-import { RENDER_SCOPE, SVG_RENDER_SCOPE } from '../react/constants';
-import type { MillionArrayProps, MillionProps, Options } from '../types';
+import type {
+  ComponentType,
+  PropsWithChildren,
+  ReactNode,
+} from 'react';
+import parse from 'html-react-parser';
+import {
+  useContainer,
+  useNearestParent,
+} from '../react/its-fine';
+import type { ArrayCache, MillionArrayProps, MillionProps, Options } from '../types';
+import { useSSRSafeId } from './utils';
 
 export { renderReactScope } from '../react/utils';
 
 let globalInfo;
 
+const isClient =  typeof window !== 'undefined' 
+const isServer = !isClient
+
+if (isClient) {
+  importSource(() => {});
+}
 export const block = <P extends MillionProps>(
   Component: ComponentType<P>,
-  options: Options = {}
+  options: Options = {},
 ) => {
   let blockFactory = globalInfo ? globalInfo.block(Component, options) : null;
 
-  const rscBoundary = options.rsc
-    ? createRSCBoundary(Component, options.svg)
-    : null;
+  const rscBoundary = options.rsc ? createRSCBoundary(Component as any) : null;
 
   function MillionBlockLoader<P extends MillionProps>(props?: P) {
-    const ref = useRef<HTMLElement>(null);
-    const patch = useRef<((props: P) => void) | null>(null);
+    const container = useContainer<HTMLElement>(); // usable when there's no parent other than the root element
+    const parentRef = useNearestParent<HTMLElement>();
 
+    const patch = useRef<((props: P) => void) | null>(null);
     const effect = useCallback(() => {
       const init = (): void => {
-        const el = ref.current;
+        const el = parentRef.current ?? container.current;
 
         if (!el) return;
 
         const currentBlock = blockFactory(props, props?.key);
 
-        globalInfo.mount(currentBlock, el, el.firstChild);
-        patch.current = (newProps: P) => {
-          globalInfo.patch(currentBlock, blockFactory(newProps, newProps.key));
-        };
+        globalInfo.mount(currentBlock, el, el.firstChild),
+          (patch.current = (newProps: P) => {
```

Comment:
hmm this doesn't look right

Reply:
oh yea i guess it's a formating thing that happened!